app/scripts/lib/util.js

```diff
@@ -299,3 +301,17 @@ export function formatTxMetaForRpcResult(txMeta) {
 
   return formattedTxMeta;
 }
+
+/**
+ * Convert a mnemonic formatted as a Uint8Array into a string.
+ *
+ * @param {Uint8Array} mnemonic - a mnemonic in Uint8Array format
+ * @returns {string} the mnemonic in string form.
+ */
+export function uint8ArrayMnemonicToString(mnemonic) {
+  const recoveredIndices = Array.from(
+    new Uint16Array(new Uint8Array(mnemonic).buffer),
+  );
+
+  return recoveredIndices.map((i) => englishWordlist[i]).join(' ');
+}
```

Comment:
Hey @adonesky1 : just curious, how is above change related to KeyringController ?

Reply:
So this helper function is used [here](https://github.com/MetaMask/metamask-extension/pull/17056/files/d11ff3886d11f58bed97b36f134745b0bc1898cc#diff-6fbff2cfe97ac01b77296ef2122c7e0a5b3ff6a84b584b4d1a87482f35eea3d6R2514). The mnemonic returned on the keyring by `this.keyringController.getKeyringsByType(KEYRING_TYPES.HD_KEY_TREE)` is now a `Uint8Array` rather than a plain text string, but the `getPrimaryKeyringMnemonic` function that is used by snaps still expects it as a string so we need this function to convert it back.